roles/db-copy/tasks/active-copy.yml

```diff
@@ -0,0 +1,137 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+---
+- name: Active-copy | Check pmon process
+  shell: "set -o pipefail; ps -ef | ( grep pmon || true ) | ( grep -i {{ db_name }} || true ) | ( grep -v grep || true ) | wc -l"
+  changed_when: false
+  register: pmon_proc
+  tags: active-duplicate
+
+- name: Active-copy | Add oratab entry
+  lineinfile:
+    path: /etc/oratab
+    regexp: '^{{ oracle_sid }}\:'
+    line: '{{ oracle_sid }}:{{ oracle_home }}:N'
+    owner: "{{ oracle_user }}"
+    group: "{{ oracle_group }}"
+  become: yes
+  tags: active-duplicate
+
+- name: Active-copy | Set random password
+  set_fact:
+    sys_pass: "{{ lookup('password', '/dev/null length=16 chars=ascii_letters,digits') }}#_"
+  tags: active-duplicate
+
+- name: Active-copy | Set sys password for primary db
```

Comment:
WIth this approach, it won't be possible to create multiple standbies, as the second run will clobber the sysdba password.  Not sure if I have a good alternative offhand though.  Maybe a way of detecting what standbies exist already and updating those?  Or something with a wallet?

Reply:
definitely if there is more than one standby to create then sys password would be overwritten, however if there are standby exist already to make sys password consistent the script should be able to connect to all standbys and parameters to connect to existing standby servers won't be a part of script execution. It might also break some connections if sys password is changed. It looks like wallet can be the only proper alternative since sys password would need to be saved somewhere securely.
Do you want to explore and develop that as part of the change or to make it as new change on top of it?

Reply:
Let's separate the changes;  for now this is fine, though it might make sense to note that creating a standby will clobber anything that relies on the current passwordfile contents.